service/tablet_allocator.cc

```diff
@@ -957,10 +959,10 @@ class load_balancer {
                 std::ranges::transform(r1, std::inserter(r1_map, r1_map.begin()), [] (tablet_replica r) {
                     return std::make_pair(r.host, r);
                 });
-                for (unsigned i = 0; i < r2.size(); i++) {
-                    auto r1_it = r1_map.find(r2[i].host);
+                for (auto& r2_replica : r2) {
```

Comment:
I _think_ this can be farther improved by using [find_first_of](https://en.cppreference.com/w/cpp/algorithm/find_first_of). 
Can you please take a closer look?

Reply:
Neither `std::find_first_of()` nor `std::ranges::find_first_of()` takes advantage of the O(1) lookup provided by `std::unordered_map`. These algorithms only require forward/input iterators from their input ranges and perform a nested linear search, resulting in O(N * M) complexity, where N is the size of the first range and M is the size of the second range.

If you want to optimize the lookup to O(N) by leveraging the unordered_map's O(1) access, we would need a different approach. However, since my goal is to maintain the current algorithm's behavior rather than optimize its performance, using `find_first_of` with its O(N * M) complexity might not be appropriate for this case.